docs/src/main/java/io/helidon/docs/se/inject/FactoryExample.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.docs.se.inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import io.helidon.common.GenericType;
+import io.helidon.service.registry.Lookup;
+import io.helidon.service.registry.Qualifier;
+import io.helidon.service.registry.Service;
+
+class FactoryExample {
+
+    record MyService() {}
+
+    // tag::snippet_1[]
+    /**
+     * Supplier service factory.
+     */
+    @Service.Singleton
+    class MyServiceProvider implements Supplier<MyService> {
+
+        @Override
+        public MyService get() {
+            return new MyService();
+        }
+    }
+    // end::snippet_1[]
+
+    // tag::snippet_2[]
+    @Service.Singleton
+    class MyServiceFactory implements Service.ServicesFactory<MyService> {
```

Comment:
`ServicesFactory` that creates a single unqualified instead does not make any sense.

Reply:
removed